### PR TITLE
Assessments tab: call assessment structure api and load into redux

### PR DESCRIPTION
**Pull request by caleybrock**
Data loaded into redux, but not used anywhere in react yet. All behind internal experiment.

Here's what redux looks like now:
screen shot 2018-05-28 at 10 19 27 pm

More about what this data looks like and why we need it here: https://github.com/code-dot-org/code-dot-org/pull/22710
(Also reminder to self to change the base of this branch to staging once other PR is merged)

### CODE REVIEW

File: apps/test/unit/templates/sectionAssessments/sectionAssessmentsReduxTest.js [hunk 1]
```diff
@@ -35,6 +36,22 @@ describe('sectionAssessmentsRedux', () => {
     });
   });
 
+  describe('setAssessmentsStructure', () => {
+    it('associates the assessment structure data to the correct script', () => {
+      const scriptId = 2;
+      const assessmentData = {139:
```
Comment (maddiedierker) [inline apps/test/unit/templates/sectionAssessments/sectionAssessmentsReduxTest.js]: nit: object structure
```
const assessmentData = {
  139: {
    id: 139,
    ...
  }
}
```

File: dashboard/app/controllers/api_controller.rb [hunk 2]
```diff
@@ -536,7 +536,6 @@ def assessments_structure
 
         case level
         when FreeResponse
-          puts level.properties
```
Comment (maddiedierker) [inline dashboard/app/controllers/api_controller.rb]: probably accidental `puts` 😄

Comment (caleybrock) [reply dashboard/app/controllers/api_controller.rb]: Yeah this was me in another PR :(

File: apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js [hunk 3]
```diff
@@ -97,3 +119,15 @@ const loadAssessmentsFromServer = (sectionId, scriptId, onComplete) => {
...
+const loadAssessmentsStructureFromServer = (scriptId, onComplete) => {
+  $.ajax({
+    url: `/dashboardapi/assessments_structure/${scriptId}`,
+    method: 'GET',
+    contentType: 'application/json;charset=UTF-8',
+  }).done(assessmentsData => {
+    onComplete(null, assessmentsData);
+  }).fail((jqXhr, status) => {
+    onComplete(status, jqXhr.responseJSON);
+  });
```
Comment (davidsbailey) [inline apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js:132]: you could return the result of `$.ajax` from this function, and then call `then` and `catch` on it

File: apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js [hunk 4]
```diff
@@ -36,6 +40,15 @@ export const asyncLoadAssessments = (sectionId, scriptId, onComplete) => {
...
     });
+
+    loadAssessmentsStructureFromServer(scriptId, (error, data) => {
+      if (error) {
+        console.error(error);
+      } else {
+        dispatch(setAssessmentsStructure(scriptId, data));
+        onComplete();
+      }
+    });
```
Comment (davidsbailey) [inline apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js:51]: no need for `asyncLoadAssessments` to take an onComplete param -- you can just return the result of `loadAssessmentsStructureFromServer`, which is a jquery deferred that also implements the Promise interface

File: apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js [hunk 5]
```diff
@@ -4,15 +4,19 @@ import {SET_SECTION} from '@cdo/apps/redux/sectionDataRedux';
  // TODO(caleybrock): define a shape for sectionAssessment data that gets stored in redux.
 const initialState = {
   assessmentsByScript: {},
+  assessmentsStructureByScript: {},
```
Comment (davidsbailey) [inline apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js:7]: What's the difference between `assessmentsByScript` and `assessmentsStructureByScript`? Adding a PropTypes shape for these (as mentioned in the TODO) could be a good way to document their structure, but ok to not block this PR on that

Comment (caleybrock) [reply apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js:7]: One is the student assessment responses and one is the assessments structure. Will document more in the future.

**Comment by caleybrock** [comment]
I'll consider promises in another PR. This reducer is still in it's early stages and can be changed pretty easily.

**Pull request merged**